File: backend/src/models/News.js
import mongoose from 'mongoose';

const newsSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true
  },
  summary: {
    type: String,
    required: true,
    trim: true
  },
  content: {
    type: String,
    required: true
  },
  source: {
    type: String,
    required: true,
    trim: true
  },
  sourceUrl: {
    type: String,
    trim: true
  },
  category: {
    type: String,
    required: true,
    trim: true
  },
  image: {
    type: String,
    trim: true
  },
  publishDate: {
    type: Date,
    default: Date.now
  },
  // 可信度相关字段
  credibilityScore: {
    type: Number,
    default: 0,
    min: 0,
    max: 100
  },
  credibilityLevel: {
    type: String,
    enum: ['Very Low', 'Low', 'Medium', 'High', 'Very High'],
    default: 'Medium'
  },
  factChecked: {
    type: Boolean,
    default: false
  },
  factCheckerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  factCheckDate: {
    type: Date
  },
  // 投票相关
  upvotes: {
    type: Number,
    default: 0
  },
  downvotes: {
    type: Number,
    default: 0
  },
  voters: [{
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User'
    },
    voteType: {
      type: String,
      enum: ['upvote', 'downvote']
    }
  }],
  // 评论相关
  commentsCount: {
    type: Number,
    default: 0
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

// 自动更新updatedAt字段
newsSchema.pre('save', function(next) {
  this.updatedAt = Date.now();
  next();
});

const News = mongoose.model('News', newsSchema);

export default News;

File: new-project/src/main/java/com/truthmoment/antifakenews/model/News.java
package com.truthmoment.antifakenews.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String summary;
    private String content;
    private String sourceUrl;
    private String category;
    private String image;
    private LocalDateTime publishDate;
    private int credibilityScore;
    private boolean factChecked;
    private int upvotes;
    private int downvotes;

    @ManyToOne
    private Source source;

    @ManyToOne
    private User factChecker;
    private LocalDateTime factCheckDate;

    @Enumerated(EnumType.STRING)
    private Source.CredibilityLevel credibilityLevel;

}